Simplified/NYPLAnnouncementBusinessLogic.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+
+private let announcementsFilename: String = "NYPLPresentedAnnouncementsList"
+
+class NYPLAnnouncementBusinessLogic {
+  static let shared = NYPLAnnouncementBusinessLogic()
+
+  private var presentedAnnouncements: Set<String> = Set<String>()
+    
+  init() {
+    restorePresentedAnnouncements()
+  }
+    
+  func presentAnnouncements(_ announcements: [Announcement]) {
+    for announcement in announcements {
+      if shouldPresentAnnouncement(id: announcement.id) {
+        DispatchQueue.main.async {
+          let vc = NYPLAnnouncementViewController(announcement: announcement)
+          NYPLRootTabBarController.shared()?.safelyPresentViewController(vc, animated: true, completion: nil)
+        }
+      }
+    }
+  }
+  
+  // MARK: - Read
+    
+  fileprivate func restorePresentedAnnouncements() {
```

Comment:
I think this and `shouldPresentAnnouncement` can be private now... I made a mistake in suggesting `fileprivate` earlier, sorry.... pretty much the only time you need `fileprivate` is if you need accessing something outside of a class in the same file, otherwise private is fine even from an `extension`.

Reply:
same for all other fileprivate stuff in this class